src/models/BaseModel.py

```diff
@@ -1,118 +1,190 @@
+
+        self.x_test = self.test_data[:, 0:self.test_data.shape[1] - 1]
+        self.y_test = self.test_data[:, self.test_data.shape[1] - 1]
+        print('Train test split complete \n')
+
+    def preprocess_data(self):
+        print('\n')
+        print('Pre-processing begins')
+        # Description: consisting of 5 features existence of words below in the description:
+        # 'excellent', 'graduate', 'immediate', 'junior', 'urgent'
+        onehot_encoded_desc_words = get_one_hot_encoded_words(self.description_feature)
+        # Contract type: One hot encoded, 3 features: part time, full time, *is empty*
+        onehot_encoded_contract_type = get_one_hot_encoded_feature(self.contract_type_feature)
+        # Contract time: One hot encoded, 3 features: permanent, contract, *is empty*
+        onehot_encoded_contract_time = get_one_hot_encoded_feature(self.contract_time_feature)
+        # Company: Binary encoded
+        binary_encoded_company = get_binary_encoded_feature(self.company_feature)
+        # Source name: Binary encoded
+        binary_encoded_source = get_binary_encoded_feature(self.source_name_feature)
+        # Town: Binary encoded
+        print('Pre-processing halfway done')
+        updated_town_feature = update_location(self.location_raw_feature,
+                                               self.cleaned_town_feature)
+        binary_encoded_town = get_binary_encoded_feature(updated_town_feature)
+        # Region: Binary encoded
+        updated_region_feature = update_location(self.location_raw_feature,
+                                                 self.cleaned_region_feature)
+        binary_encoded_region = get_binary_encoded_feature(updated_region_feature)
+        # Job titles and modifiers: Binary encoded
+
+        processed_job_titles, processed_job_modifiers = get_stemmed_sentences(
+            self.title_feature
+        )
+        binary_encoded_job_titles = get_binary_encoded_feature(processed_job_titles)
+        binary_encoded_job_modifiers = get_binary_encoded_feature(processed_job_modifiers)
+        # Job category: Binary encoded
+        processed_category_feature = remove_sub_string('Jobs', self.category_feature)
+        binary_encoded_categories = get_binary_encoded_feature(processed_category_feature)
+
+        concatanated_features = np.concatenate((
+                    onehot_encoded_desc_words,
+                    onehot_encoded_contract_type,
+                    onehot_encoded_contract_time,
+                    binary_encoded_company,
+                    binary_encoded_source,
+                    binary_encoded_town,
+                    binary_encoded_region,
+                    binary_encoded_job_titles,
+                    binary_encoded_job_modifiers,
+                    binary_encoded_categories,
+                    [pandas_vector_to_list(self.salary_feature)]
+        ))
+        concatanated_features = np.transpose(concatanated_features)
+
```

Comment:
Also wanted to create a list of tuples to see how long(columns) each feature holds